src/render.rs

```diff
@@ -53,15 +54,23 @@ pub struct Renderer<'a> {
     context: Value,
     tera: &'a Tera,
     for_loops: Vec<ForLoop>,
+    should_escape: bool,
 }
 
 impl<'a> Renderer<'a> {
     pub fn new(tpl: &'a Template, tera: &'a Tera, context: Value) -> Renderer<'a> {
+        let mut should_escape = false;
+        for ext in &tera.autoescape_extensions {
+            if tpl.name.ends_with(ext) {
```

Comment:
Not too convinced by using Path::extension(), it only gets the bit after the last dot, which fails for files with several dots, ie if someone wants to escape `.php.html` or whatever. Tempted to just add the `.` for the default and ler user handle that themselves